README.md

```diff
@@ -5,4 +5,43 @@
 [![Coverage Status](https://img.shields.io/coveralls/yast/yast-bootloader.svg)](https://coveralls.io/r/yast/yast-bootloader?branch=master)
 [![inline docs](http://inch-ci.org/github/yast/yast-bootloader.svg?branch=master)](http://inch-ci.org/github/yast/yast-bootloader)
 
+## Goal
+
+This module has two main responsibilities:
+
+1. Proposing a bootable configuration so even beginners who've never heard about
+   booting can get a bootable distribution out of the box without any interaction.
+
+2. Letting advanced users who already know what they want to achieve
+   to edit an existing configuration or a proposal.
+
 Check our list of [supported scenarios.](SUPPORTED_SCENARIOS.md)
+
+## Development
+
+### High Level Overview
+
+The bootloader module consists of two more or less separated components.
+The **UI** including dialogs, widgets and similar, and
+the **backend** that is responsible for reading, writing, proposing.
+
+### Backend
+
+![overview picture](doc/bootloader_backend.svg)
```

Comment:
SVG on GitHub is tricky: https://github.com/github/markup/issues/556
Maybe you want to add a note:

> To get to the clickable links, right-click the diagram and choose "Display Image" (Firefox) or "Open Image in New Tab" (Chromium).

Reply:
Also note a trick with raw.github.com/...foo.svg?sanitize=true : https://stackoverflow.com/questions/13808020/include-an-svg-hosted-on-github-in-markdown